projects/ngx-d3timeline/src/lib/axis.service.ts

```diff
@@ -1,26 +1,34 @@
 import { Injectable } from '@angular/core';
 import { BehaviorSubject, combineLatest } from 'rxjs';
-import { map, pluck } from 'rxjs/operators';
+import { map, pluck, switchMap } from 'rxjs/operators';
 import { TimelineEvent } from './timeline-event';
 import { ResourcesAxisViewModel } from './axis/resources-axis-view-model';
 import { ViewService } from './view/view.service';
 import { TimeAxisViewModel } from './axis/time-axis-view-model';
+import { EventService } from './event.service';
 
 @Injectable({ providedIn: 'root' })
 export class AxisService {
   private dataSubject = new BehaviorSubject<TimelineEvent[]>(null);
 
   axis$ = combineLatest(this.dataSubject, this.viewService.view$).pipe(
-    map(([data, timelineView]) => ({
-      resourceAxisVm: new ResourcesAxisViewModel(data, timelineView),
-      timeAxisVm: new TimeAxisViewModel(data, timelineView)
-    }))
+    switchMap(([data, timelineView]) =>
```

Comment:
Doesn't need changing here/now, but I think we can refer to instances of TimelineView as just view. TimelineView differentiates from other View types, but I don't think making the distinction is nearly as important on the instance names.